src/webgl/p5.Camera.js

```diff
@@ -1717,26 +1720,25 @@ p5.Camera.prototype._getLocalAxes = function() {
  * @param {Number} dRadius change in radius
  */
 p5.Camera.prototype._orbit = function(dTheta, dPhi, dRadius) {
+  // Calculate the vector and its magnitude from the center to the viewpoint
   const diffX = this.eyeX - this.centerX;
   const diffY = this.eyeY - this.centerY;
   const diffZ = this.eyeZ - this.centerZ;
-
-  // get spherical coorinates for current camera position about origin
-  let camRadius = Math.sqrt(diffX * diffX + diffY * diffY + diffZ * diffZ);
-  // from https://github.com/mrdoob/three.js/blob/dev/src/math/Spherical.js#L72-L73
-  let camTheta = Math.atan2(diffX, diffZ); // equatorial angle
-  let camPhi = Math.acos(Math.max(-1, Math.min(1, diffY / camRadius))); // polar angle
-
-  let newUpY = this.upY > 0 ? 1 : -1;
-  // add change according to the direction of newupY
-  camTheta += newUpY * dTheta;
-  camPhi += newUpY * dPhi;
-  // if camPhi becomes >= PI or <= 0,
-  // upY of camera need to be flipped to the other side
-  if (camPhi <= 0 || camPhi >= Math.PI) {
-    newUpY *= -1;
-  }
-
+  let camRadius = Math.hypot(diffX, diffY, diffZ);
+  // front vector. unit vector from center to eye.
+  const front = new p5.Vector(diffX, diffY, diffZ).normalize();
+  // up vector. normalized camera's unit vector.
+  const up = new p5.Vector(this.upX, this.upY, this.upZ).normalize(); // y-axis
+  // side vector. Right when viewed from the front
+  const side = new p5.Vector.cross(up, front).normalize(); // x-axis
+  // verticalVector. projection of front vector.
+  const vertical = new p5.Vector.cross(side, up).normalize(); // z-axis
+  // calculate camPhi (Rise angle when up is regarded as 0)
+  let camPhi = Math.acos(Math.max(-1, Math.min(1, p5.Vector.dot(front, up))));
```

Comment:
It might be worth adding a comment that this is like doing `Math.abs(front.angleBetween(up))` (but more efficiently, since that internally does an extra cross product to get the sign of the angle.)

Reply:
I looked it up and found that angleBetween does the same calculation! Fix!

Reply:
I would like the comment to be "Find the angle between the up vector and the front vector".
Or rather, when I looked it up, it was the same calculation as angleBetween(), so I'll rewrite it that way.